.github/workflows/release.yml

```diff
@@ -75,6 +75,32 @@ jobs:
         bin/docker-retag-all $TAG main
         bin/docker-push main
   # todo: Keep in sync with `kind_integration.yml`
+  windows_static_cli_tests:
+    name: Static CLI tests (windows)
+    runs-on: windows-latest
+    needs: [docker_push]
+    steps:
+    - name: Checkout code
+      # actions/checkout@v2
+      uses: actions/checkout@722adc6
+    - name: Try to load cached Go modules
+      # actions/cache@v1.1.2
+      uses: actions/cache@70655ec
+      with:
+        path: ~/go/pkg/mod
+        key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+        restore-keys: |
+          ${{ runner.os }}-go-
+    - name: Install linkerd CLI
+      id: install_cli
+      run: |
+        $image="gcr.io/linkerd-io/cli-bin:$TAG"
```

Comment:
Each job needs to set it's own environment variables. `$TAG` hasn't been set in `windows_static_cli_tests` yet, so you'll need another copy of the `Set environment variables from scripts` step in this job!

Reply:
Ah, @alpeb and I were expecting them to be present throughout the workflow and he is testing it. We previously went with the main tag as its also pushed everytime but then switched over to $TAG.

The problem with running is the script is that its a windows machine and that bash script would not work! :\

Reply:
@Pothulapati note though that [you can use bash in windows as well](https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#using-a-specific-shell). Worth a shot if the TAG env var is not available :wink:

Reply:
I tested in a fork using this minimal workflow:
```yaml
name: Release
on:
  push:
    branches:
    - master
env:
  GH_ANNOTATION: true
  TAG: ""
jobs:
  set_env_var:
    name: Setting TAG
    runs-on: ubuntu-18.04
    steps:
    - name: Checkout code
      # actions/checkout@v2
      uses: actions/checkout@722adc6
    - name: Set environment variables from scripts
      run: |
        . bin/_tag.sh
        echo ::set-env name=TAG::$(CI_FORCE_CLEAN=1 bin/root-tag)
  get_env_var:
    name: Getting TAG (linux)
    needs: [set_env_var]
    runs-on: ubuntu-18.04
    steps:
    - name: Check TAG using bash
      shell: bash
      run: |
        echo "TAG:"
        echo $TAG
  windows_static_cli_tests:
    name: Getting TAG (windows)
    needs: [set_env_var]
    runs-on: windows-latest
    steps:
    - name: Check TAG using bash
      shell: bash
      run: |
        echo "TAG:"
        echo $TAG
    - name: Check TAG using PS
      run: |
        echo "TAG:"
        echo $env:TAG
    - name: Checkout code
      # actions/checkout@v2
      uses: actions/checkout@722adc6
    - name: Run bin/root-tag
      shell: bash
      run: bin/root-tag
```
Results [here](https://github.com/alpeb/linkerd2/runs/812705443?check_suite_focus=true).

As @Pothulapati noticed before, global env vars are only for static values; attempting to change them via `echo ::set-env name=TAG::...` doesn't work. That only changes the value for the current job and the value isn't carried over the following jobs, regardless of them using linux or windows (PS or bash).
However, if we set the shell to be `bash`, we can run `bin/root-tag` to retrieve the tag!